python/pyspark/worker.py

```diff
@@ -357,8 +357,14 @@ def map_batch(batch):
             num_output_rows = 0
             for result_batch, result_type in result_iter:
                 num_output_rows += len(result_batch)
-                assert is_map_iter or num_output_rows <= num_input_rows[0], \
-                    "Pandas MAP_ITER UDF outputted more rows than input rows."
+
+                if is_scalar_iter and num_output_rows > num_input_rows[0]:
+                    raise RuntimeError(
+                        "The length of each output series (or frame) in Scalar iterator pandas "
+                        "UDF should be the same with the input's; however, the length of output "
+                        "series (or frame) was %d and the length of the input's was %d." % (
+                            num_output_rows, num_input_rows[0]))
+
```

Comment:
message: Pandas MAP_ITER UDF outputted more rows than input rows.

Reply:
Pandas MAP_ITER UDF does not have any restriction on lengths :-).